### ISSUE TITLE
"TypeError: Cannot convert undefined or null to object" when running tests in the node-libquassel project

### ISSUE THREAD

**Issue by binsky08** [magne4000/node-qtdatastream#13]
When running `npm run test` in the corresponding [node-libquassel](https://github.com/magne4000/node-libquassel) project, the test fails due to an error in this lib.

```
> libquassel@4.0.0 test
> tape -r @babel/register test/test.js

TAP version 13
# setup
ok 1 setup event received
ok 2 setupfailed event received
ok 3 setupok event received
# login
ok 4 loginfailed event received
ok 5 login event received
ok 6 bufferview.ids event received
ok 7 ignorelist event received
ok 8 aliases event received
# identity
ok 9 identity.new event received
ok 10 should be strictly equal
ok 11 should be strictly equal
ok 12 should be deeply equivalent
ok 13 should be strictly equal
ok 14 should be strictly equal
# network
/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:795
      const keys = Object.keys(this.__obj);
                          ^

TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at QMap.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:795:27)
    at QVariant.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:1085:41)
    at QMap.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:802:50)
    at QUserType.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:918:79)
    at QVariant.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:1085:41)
    at QList.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:636:35)
    at QVariant.toBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/types.js:1085:41)
    at Function.getBuffer (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/transform.js:178:46)
    at WriteTransform._transform (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/transform.js:186:35)
    at WriteTransform.Transform._write (node:internal/streams/transform:171:8)
    at writeOrBuffer (node:internal/streams/writable:570:12)
    at _write (node:internal/streams/writable:499:10)
    at WriteTransform.Writable.write (node:internal/streams/writable:508:10)
    at Socket.write (/home/timo/Entwicklung/node-libquassel/node_modules/qtdatastream/src/socket.js:58:23)
    at Core.createNetwork (/home/timo/Entwicklung/node-libquassel/src/request.js:79:23)
    at Test.<anonymous> (/home/timo/Entwicklung/node-libquassel/test/test.js:122:16)
    at Test.run (/home/timo/Entwicklung/node-libquassel/node_modules/tape/lib/test.js:126:28)
    at Immediate.next (/home/timo/Entwicklung/node-libquassel/node_modules/tape/lib/results.js:158:7)
    at processImmediate (node:internal/timers:483:21)

Node.js v22.4.1
```

**Fix:**

If you set [keys](https://github.com/magne4000/node-qtdatastream/blob/81b68829b2687fff3297b3115910898010ed6dd2/src/types.js#L795) to an empty array when this.__obj is undefined, the test goes green and the tested functionality works. In this case, the creation of a test network in the connected quassel-core instance.

**Comment by magne4000** [comment]
Published in `v1.1.1`. You could force update `node-qtdatastream` by deleting your npm lock file.

**Comment by binsky08** [comment]
Thanks :)
